### PR TITLE
CIS Page Rule development

### PR DESCRIPTION
**Pull request by KumarGanesanIBM**

### CODE REVIEW

File: ibm/resource_ibm_cis_page_rule.go [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+	cisClient, err := meta.(ClientSession).CisPageRuleClientSession()
+	if err != nil {
+		return err
+	}
+
+	ruleID, zoneID, crn, _ := convertTfToCisThreeVar(d.Id())
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneID = core.StringPtr(zoneID)
+
+	if d.HasChange(cisPageRuleTargets) ||
```
Comment (kavya498) [inline ibm/resource_ibm_cis_page_rule.go:241]: Priority and status doesn't have update functionality?

File: ibm/resource_ibm_cis_page_rule.go [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+		opt.SetTargets(targets)
+		opt.SetActions(actions)
+		if value, ok := d.GetOk(cisPageRulePriority); ok {
+			opt.SetPriority(int64(value.(int)))
+		}
+		if value, ok := d.GetOk(cisPageRuleStatus); ok {
+			opt.SetStatus(value.(string))
+		}
+
+		_, response, err := cisClient.UpdatePageRule(opt)
```
Comment (kavya498) [inline ibm/resource_ibm_cis_page_rule.go:257]: There are two features `update` and `change` in api docs are they both same? If not where is it handled in resource?

File: ibm/resource_ibm_cis_page_rule_test.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+		targets {
+		  target = "url"
+		  constraint {
+			operator = "matches"
+			value    = "%[1]s"
+		  }
+		}
+		status   = "active"
+		priority = 1
+		actions {
```
Comment (kavya498) [inline ibm/resource_ibm_cis_page_rule_test.go:149]: can more actions be added to testcase?

File: website/docs/r/cis_page_rule.html.markdown [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    - 'Origin error page pass-through': `origin_error_page_pass_thru`,
+    - 'Response buffering': `response_buffering`,
+    - 'Image size optimization': `image_size_optimization`,
+    - 'Script load optimization': `script_load_optimization`,
+    - 'True client IP header': `true_client_ip_header`,
+    - 'Sort query string': `sort_query_string_for_cache`,
+  - `value` - (Required, string) The Values for corresponding actions are below,
+    - `always_online` - valid values: `on`, `off`.
+    - `ssl` - valid values: `off`, `flexible`, `full`, `strict`, `origin_pull`.
+    - `browser_cache_ttl` - valid values: `0`, `1800`, `3600`, `7200`, `10800`, `14400`, `18000`, `28800`, `43200`, `57600`, `72000`, `86400`, `172800`, `259200`, `345600`, `432000`, `691200`, `1382400`, `2073600`, `2678400`, `5356800`, `16070400`, `31536000`.
```
Comment (kavya498) [inline website/docs/r/cis_page_rule.html.markdown:85]: few are int and few are string..so, can you please mention type of attribute..

File: website/docs/r/cis_page_rule.html.markdown [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  Provides a IBM CIS Page Rule resource.
+---
+
+# ibm_cis_page_rule
+
+Provides a IBM CIS Page Rule resource. This resource is associated with an IBM Cloud Internet Services instance and a CIS Domain resource. It allows to create, update, delete page rules of a domain of a CIS instance
+
+## Example Usage
+
+```hcl
```
Comment (kavya498) [inline website/docs/r/cis_page_rule.html.markdown:15]: can the example for forwarding url and minify be added as they are different when compared to others

File: ibm/data_source_ibm_cis_page_rules.go [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+				Description:      "DNS Zone ID",
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisPageRules: {
+				Type:        schema.TypeList,
+				Description: "Collection of page rules detail",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
```
Comment (kavya498) [inline ibm/data_source_ibm_cis_page_rules.go:41]: "id" can be ignored as it is not set in read method..

File: ibm/resource_ibm_cis_page_rule.go (hunk 1 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_page_rule.go:242]: added check for `priority` and `status`. But `priority` field is not supported as of now.

File: ibm/resource_ibm_cis_page_rule.go (hunk 2 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_page_rule.go:260]: both does the same.

File: ibm/resource_ibm_cis_page_rule_test.go (hunk 3 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_page_rule_test.go:149]: added one more action.

File: website/docs/r/cis_page_rule.html.markdown (hunk 5 above)
Comment (KumarGanesanIBM) [reply website/docs/r/cis_page_rule.html.markdown:15]: minify is not supported yet.

Comment (kavya498) [reply website/docs/r/cis_page_rule.html.markdown:15]: but i see minify in code

Comment (kavya498) [reply website/docs/r/cis_page_rule.html.markdown:15]: is the resource not supporting or api?

Comment (KumarGanesanIBM) [reply website/docs/r/cis_page_rule.html.markdown:15]: I removed the `minify` from validator.

Comment (kavya498) [reply website/docs/r/cis_page_rule.html.markdown:15]: If the api supports please add minify feature as well in the resource..

Comment (KumarGanesanIBM) [reply website/docs/r/cis_page_rule.html.markdown:15]: API does not support as of now.

**Pull request merged**